shared/js/ui/models/grade-scorecard.es6.js

```diff
@@ -0,0 +1,75 @@
+const Parent = window.DDG.base.Model
+
+function GradeScorecard (attrs) {
+  Parent.call(this, attrs)
+
+  this.reasons = this._getReasons()
+}
+
+GradeScorecard.prototype = window.$.extend({},
+  Parent.prototype,
+  {
+    modelName: 'gradeScorecard',
+
+    _getReasons: function () {
+      const site = this.site
+      let reasons = []
+
+      // grab all the data from the site to create
+      // a list of reasons behind the grade
+
+      // encryption status
+      let httpsStatusText = site.httpsStatusText
+      if (httpsStatusText) {
+        let connectionMsg = 'Unencrypted'
+        let modifier = 'bad'
+
+        if (httpsStatusText === 'Secure') {
+          connectionMsg = 'Encrypted'
+          modifier = 'good'
+        }
+
+        reasons.push({
+          modifier,
+          msg: `${connectionMsg} Connection`
```

Comment:
I definitely see where you're coming with this, but doing this will require the templates to be able to differentiate between different kinds of reasons. This means spreading the idea of what a "reason" is between multiple entities, which probably makes this model unnecessary.

Wouldn't you say it's better to keep this model as the sole source of truth for what the scorecard reasons are, and keep the template as dumb and logic-less as possible?

Reply:
(I'm not strongly attached to this BTW, if you're still not convinced I'll do it your way.)

Reply:
That said, by the same token I should move the grades to this model as well. Let's see what we decide regarding this model generally, then move them or not.

Reply:
Sounds like a template mixin could be the single source of truth for message mappings then? Saying this because at some point if/when we internationalize, i want all of that logic strictly in the templates and not spread out all over the place in models and views.

Reply:
OK, I buy that.

I may end up shifting this entire thing in the template then. The vast majority of this model is mapping values from the site model to a message.